src/Notifications/data/slice.js

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+
+export const IDLE = 'idle';
+export const LOADING = 'loading';
+export const LOADED = 'loaded';
+export const FAILED = 'failed';
+export const DENIED = 'denied';
+
+const initialState = {
+  notificationStatus: 'idle',
+  appName: 'reminders',
+  appsId: [],
+  apps: {},
+  notifications: {},
+  tabsCount: {},
+  showNotificationTray: false,
+  pagination: {
+    count: 10,
+    numPages: 1,
+    currentPage: 1,
+    nextPage: null,
+  },
+};
+const slice = createSlice({
+  name: 'notifications',
+  initialState,
+  reducers: {
+    fetchNotificationDenied: (state, { payload }) => {
+      state.appName = payload.appName;
+      state.notificationStatus = DENIED;
+    },
+    fetchNotificationFailure: (state, { payload }) => {
+      state.appName = payload.appName;
+      state.notificationStatus = FAILED;
+    },
+    fetchNotificationRequest: (state, { payload }) => {
+      if (state.appName !== payload.appName) { state.apps[payload.appName] = []; }
+      state.appName = payload.appName;
+      state.notificationStatus = LOADING;
+    },
+    fetchNotificationSuccess: (state, { payload }) => {
+      const { notifications, numPages, currentPage } = payload;
+      const newNotificationIds = notifications.map(notification => notification.id.toString());
+      const existingNotificationIds = state.apps[state.appName];
+      const notificationsKeyValuePair = notifications.reduce((acc, obj) => { acc[obj.id] = obj; return acc; }, {});
+      const currentAppCount = state.tabsCount[state.appName];
+
+      state.apps[state.appName] = Array.from(new Set([...existingNotificationIds, ...newNotificationIds]));
+      state.notifications = { ...state.notifications, ...notificationsKeyValuePair };
+      state.tabsCount.count -= currentAppCount;
+      state.tabsCount[state.appName] = 0;
+      state.notificationStatus = LOADED;
+      state.pagination.numPages = numPages;
+      state.pagination.currentPage = currentPage;
+    },
+    fetchNotificationsCountDenied: (state) => {
+      state.notificationStatus = DENIED;
+    },
+    fetchNotificationsCountFailure: (state) => {
+      state.notificationStatus = FAILED;
+    },
+    fetchNotificationsCountRequest: (state) => {
+      state.notificationStatus = LOADING;
+    },
+    fetchNotificationsCountSuccess: (state, { payload }) => {
+      const { countByAppName, count, showNotificationTray } = payload;
+      state.tabsCount = { count, ...countByAppName };
+      state.appsId = Object.keys(countByAppName);
+      state.apps = Object.fromEntries(Object.keys(countByAppName).map(key => [key, []]));
+      state.showNotificationTray = showNotificationTray;
+      state.notificationStatus = LOADED;
```

Comment:
pass the normalized data in the slice and assign it to the required state. Please avoid any unnecessary calculation in slice.